tscat_gui/__init__.py

```diff
@@ -9,48 +9,55 @@
 from PySide2 import QtGui
 
 import tscat
+
 from .model import CatalogueModel
 
-from .edit import CatalogueEditWidget
+from .edit import EntityEditWidget
+
+from .undo import stack
+
+from .utils.helper import get_entity_from_uuid_safe
 
 
 class TSCatGUI(QtWidgets.QWidget):
     def __init__(self, parent=None):
         super().__init__(parent)
 
+        stack.setup(self)
```

Comment:
You didn't make this scoped? A global undo unique stack is really dangerous while we can spawn several Edit widgets from the same process.

Reply:
No it's not safe. I haven't had a good idea until now. Having an instanced stack would basically make every widget take an additional argument, even it is only for passing on to other objects/widgets. The stack (or the main-widget because the stack needs access to refresh it) needs to be accessible from everywhere where commands are created.

Maybe we can do something with signal-slots...

Reply:
Passing undo stack to all widgets ctor looks ok to me, so to parent widget creates its own undo stack instance and share it with children.